src/Renderer/Shader/PointsVS.glsl

```diff
@@ -4,32 +4,35 @@ precision highp int;
 #include <logdepthbuf_pars_vertex>
 #define EPSILON 1e-6
 
-uniform float size;
-uniform float scale;
+attribute vec3 position;
 uniform mat4 projectionMatrix;
 uniform mat4 modelViewMatrix;
+
+uniform float size;
 uniform vec2 resolution;
-uniform bool pickingMode;
 
-attribute vec4 unique_id;
+uniform bool pickingMode;
+uniform float opacity;
+uniform vec4 overlayColor;
 attribute vec3 color;
-attribute vec3 position;
+attribute vec4 unique_id;
 
 varying vec4 vColor;
 
 void main() {
     if (pickingMode) {
         vColor = unique_id;
     } else {
-        vColor = vec4(color, 1.0);
+        vColor = vec4(mix(color, overlayColor.rgb, overlayColor.a), opacity);
     }
+
     vec4 mvPosition = modelViewMatrix * vec4( position, 1.0 );
-    mat4 projModelViewMatrix = projectionMatrix * modelViewMatrix;
-    gl_Position = projModelViewMatrix* vec4( position, 1.0);
+    gl_Position = projectionMatrix * mvPosition;
 
-    if (size > 0.01) {
+    if (size > 0.) {
         gl_PointSize = size;
     } else {
+        // automatic sizing
```

Comment:
I may be wrong, but we could even only rely on `resolution.y` in fact, by applying a smaller factor (0.003 instead of -0.05 ?)

Reply:
Looking [at the sse code](https://github.com/iTowns/itowns/blob/6b45cf1d4ef863a8ecbf9d884ed545520507f634/src/Process/PointCloudProcessing.js#L55-L67), I am tempted to propose  (do we really need to clamp?)
```
gl_PointSize = (size <0.) ? (size / gl_Position.w) : size;
```
with size being a float uniform set to :
- positive: the sprite size in pixels  or 
- negative: `- context.camera.preSSE * pointSpacing` (possibly multiplied by a user provided tuning factor with a default to 1)

I agree with your proposal for ditching `mvPosition` and passing in a single matrix
(I do not get your second comment)

Reply:
Sorry for the bad explanation. I found that `gl_Position = clamp(0.003 * resolution.y, 3.0, 10.0);` gives a pretty good result. If we are ok with this default value, which does not depend from the point position, we could remove the clamp from the shader, and only compute a default size once in javascript. And then we would only have `gl_Position = size;`.

Reply:
@zarov, @mbredif: did you test with different pointclouds (to avoid picking a solution that only works well for one example)?

Reply:
Just one little remark while passing by: as the calculus is quite simple from the computer POV, we should really aim for readability here. I'm not saying the current code is good regarding this, but that might be a reason to avoid one-liner or to keep the calculus in shader... I expect neither the cost of calculating here nor uploading an additional uniform to really matter here anyway.

For this kind of work, the best way is to test and make screenshots to show reviewers imo :-)

Reply:
@peppsac no, but from what I understand: there is no need to compute a varying default point size. The only thing we want is a good ratio to the size of the canvas. If the user wants to set a specific and varying size, he still can. But I agree that this formula might not be the one we want, and other use-cases need to be tested. 

@autra I agree, and by moving to `gl_Position = size;` I think it gives the most readability.

Reply:
For the moment, I just propose a clean up (as in "removing dead code" : bouh! :) ) and add a slight generalization. That will turn
```
        // automatic sizing
        float pointSize = 1.0;
        float slope = tan(1.0 / 2.0);
        float projFactor =  -0.5 * resolution.y / (slope * mvPosition.z);

        float z = min(0.5 * -gl_Position.z / gl_Position.w, 1.0);
        gl_PointSize = max(3.0, min(10.0, 0.05 * projFactor));
```
into
```
        gl_PointSize = clamp(size * scale / gl_Position.w, 3.0, 10.0);
```
or even
```
        gl_PointSize = size * scale / gl_Position.w;
```
The previous automatic behavior would be matched if  `size * scale = (0.05 * -0.5 / tan(1.0 / 2.0)) * resolution.y` (for instance `size = -0.04576219304`, wherever that magic value comes from, and `scale = resolution.y`). That simplification would be almost a perfect match with the [THREE.PointsMaterial code](https://github.com/mrdoob/three.js/blob/master/src/renderers/shaders/ShaderLib/points_vert.glsl#L17-L21)

(@zarov I agree with @peppsac that we should not completely remove automatic sizing)

Reply:
Well if you want to remove dead code only, remove only the `pointSize` line and the `float z` which are unused. I'd rather commit a thorough solution if we want to change things inside.